PR: Restructure console menus

juanis2112 opened the pull request:
<!--- Make sure to read the Contributing Guidelines:                   --->
<!--- https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md --->
<!--- and follow PEP 8, PEP 257 and Spyder's code style:               --->
<!--- https://github.com/spyder-ide/spyder/wiki/Dev:-Coding-Style      --->

## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

<!--- Explain what you've done and why --->
Rearranged IPython console options menu and IPython main console options menu. 

![cli](https://user-images.githubusercontent.com/16781833/51806633-31cdb580-224a-11e9-9c99-c3337a47ae46.png)

![top](https://user-images.githubusercontent.com/16781833/51806635-32fee280-224a-11e9-854b-1d3bad9ff931.png)

### Issue(s) Resolved

<!--- List the issue(s) below, in the form "Fixes #1234"; one per line --->

Fixes #8478 


### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: Juanis2112

<!--- Thanks for your help making Spyder better for everyone! --->

pep8speaks commented:
Hello @juanis2112! Thanks for submitting the PR.

 - In the file [`spyder/plugins/ipythonconsole/plugin.py`](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py), following are the PEP8 issues :

> [Line 30:80](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L30): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 31:80](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L31): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 47:1](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L47): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 739:24](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L739): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 750:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L750): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 751:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L751): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 752:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L752): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 755:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L755): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 756:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L756): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 757:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L757): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 791:1](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/plugin.py#L791): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

 - In the file [`spyder/plugins/ipythonconsole/widgets/client.py`](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py), following are the PEP8 issues :

> [Line 48:62](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L48): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 383:38](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L383): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 384:38](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L384): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 385:1](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L385): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 413:32](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:34](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:36](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:38](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:40](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:42](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:44](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:46](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:48](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:50](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 413:53](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L413): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 415:32](https://github.com/spyder-ide/spyder/blob/a92393a520173363e9aa96884fc82b478183e712/spyder/plugins/ipythonconsole/widgets/client.py#L415): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

ccordoba12 commented:
@juanis2112, to do the rebase you need to run these commands

```
git checkout master
git fetch upstream
git merge upstream/master
git checkout Solve_issue8478
git rebase master
```

CAM-Gerlach commented:
> git merge upstream/master

Is there a reason to use this over ``git pull upstream master``? I'm just curious if I've been doing it wrong all these years, lol.

CAM-Gerlach commented:
See my comment on #8478 .

CAM-Gerlach commented on spyder/plugins/ipythonconsole/widgets/client.py:
```diff
@@ -379,13 +379,17 @@ def get_kernel(self):
         env_action = create_action(
                         self,
-                        _("Show environment variables"),
+                        _("Show env vars"),
```
``"Show environment variables"`` is clearer to those who may not know the meaning of the abbreviations (non-natives, newer users, etc) and still shorter than ``Connect to an existing kernel`` so it doesn't make the menu any wider anyway.

ccordoba12 commented:
@juanis2112, there are still some pep8 issues to solve.

CAM-Gerlach commented:
Also, once its final, @juanis2112 should update the screenshot in the OP.

dalthviz commented:
@ccordoba12 @CAM-Gerlach I have updated the screenshot of the menu, what do you think?

ccordoba12 commented:
> I have updated the screenshot of the menu, what do you think?

![cli](https://user-images.githubusercontent.com/16781833/51806633-31cdb580-224a-11e9-9c99-c3337a47ae46.png)

This menu is done for me, so please don't touch it any longer.

> ![top](https://user-images.githubusercontent.com/16781833/51806635-32fee280-224a-11e9-854b-1d3bad9ff931.png)

We agreed with @CAM-Gerlach to also have here the `New special console` submenu below `New console`, so that's still missing. @juanis2112, please address that here.

----

By the way, great work @juanis2112 and @dalthviz!! This looks really good!

juanis2112 commented:
> > I have updated the screenshot of the menu, what do you think?
> 
> ![cli](https://user-images.githubusercontent.com/16781833/51806633-31cdb580-224a-11e9-9c99-c3337a47ae46.png)
> 
> This menu is done for me, so please don't touch it any longer.
> 
> ![top](https://user-images.githubusercontent.com/16781833/51806635-32fee280-224a-11e9-854b-1d3bad9ff931.png)
> 
> We agreed with @CAM-Gerlach to also have here the `New special console` submenu below `New console`, so that's still missing. @juanis2112, please address that here.
> 
> By the way, great work @juanis2112 and @dalthviz!! This looks really good!

Thanks @ccordoba12. I just added the special console menu in the main consoles menu.

![screen shot 2019-01-30 at 12 14 01 pm](https://user-images.githubusercontent.com/18587879/51999554-08e73380-2489-11e9-9b54-a5643f892ab1.png)

ccordoba12 commented:
Thanks @juanis2112!

ccordoba12 commented:
> However, it seems like the consensus is firmly against adding them

Well, we can always revise our decision. @bcolsen, @jnsebgosselin, @jitseniesen, do you have something to say about this?

jitseniesen commented:
I am also against adding the items. Environment variables seems very specialized. The path less so but it still feels out of place to me.

ccordoba12 commented:
Ok, I think we reached a consensus. Thanks @jitseniesen for the input.

dalthviz commented:
@juanis2112 could you fix the conflicts to merge this? Thanks :)

ccordoba12 commented:
@dalthviz, please merge if this is ready.
